src/core/mac/data_poll_sender.hpp

```diff
@@ -171,8 +171,11 @@ class DataPollSender : public InstanceLocator
      * the received frame header and if it is set, data poll sender will send an immediate data poll to retrieve the
      * pending frame.
      *
+     * If enhanced keep-alive is supported, this method also resets keep-alive timer if the frame is an Ack.
+     *
+     * @param[in]   aFrame
      */
-    void CheckFramePending(Mac::RxFrame &aFrame);
+    void ProcessFrame(const Mac::RxFrame &aFrame);
```

Comment:
I think it would be useful to update the first line of the doxy comment here, as I'm not sure the usage is totally intuitive. It seems ``ProcessFrame`` is only called to notify the data poll sender in 2 situations:
1. Receiving a Data Frame indirectly, in both 1.1 and 1.2. (if FP is set we send another data poll.)
1. Receiving an Ack for a sent data frame in 1.2 only. (If FP we send a data poll, if not we reset the keepalive timer for sending next poll.)

Reply:
Thank you for reviewing. Would you like to contribute a GitHub suggestion  (e.g. https://github.com/openthread/openthread/pull/4334#discussion_r388277192) for the doxy?

Reply:
@CiaranWoodward could you review the updated doc string?

Reply:
Apologies for not getting around to your request, but new text looks good to me!